### PR TITLE
CA-204240: Add more vif-configure-ip command paramters check

### PR DESCRIPTION
**Pull request by cheng-z**
Signed-off-by: chengz <cheng.zhang@citrix.com>

### CODE REVIEW

**Comment by cheng-z** [comment]
Hi @jonludlam 
Thanks a lot for your suggestion.
Please help to review my updated pull request.
Thanks.

File: ocaml/xapi/cli_operations.ml [hunk 1]
```diff
@@ -1579,14 +1579,16 @@ let vif_configure_ipv4 printer rpc session_id params =
 	let mode = Record_util.vif_ipv4_configuration_mode_of_string (List.assoc "mode" params) in
 	let address = List.assoc_default "address" params "" in
 	let gateway = List.assoc_default "gateway" params "" in
-	let () = Client.VIF.configure_ipv4 rpc session_id vif mode address gateway in ()
+	if mode = `Static && address = "" then failwith "Required parameter not found: address"
+	else let () = Client.VIF.configure_ipv4 rpc session_id vif mode address gateway in ()
```
Comment (jonludlam) [inline ocaml/xapi/cli_operations.ml]: This is slightly odd syntactically. Better to do:

```
if mode = `Static && address = "" then failwith "Required parameter not found: address";
Client.VIF.configure_ipv4 rpc session_id vif mode address gateway
```

**Comment by cheng-z** [comment]
Hi @jonludlam 
Thanks for your comment, I already updated the pull request and please help to review and see if we can merge it or not.
Thanks.

**Comment by jonludlam** [comment]
Don't think that failure is real - not something I've seen before though:

```
Broadcast message from root@testing-gce-199dac86-24c9-4ff0-bad7-b45c7f188eec
	(unknown) at 3:12 ...
The system is going down for power off NOW!
Pulling repository docker.io/xenserver/xenserver-build-env
unexpected EOF
```

**Comment by jonludlam** [comment]
I've restarted it.

**Comment by jonludlam** [comment]
OK, everything working now!

**Pull request merged**